packages/gatsby/src/commands/serve.ts

```diff
@@ -118,6 +119,8 @@ module.exports = async (program: IServeProgram): Promise<void> => {
   router.use(compression())
   router.use(express.static(`public`, { dotfiles: `allow` }))
 
+  router.use(configureTrailingSlash(trailingSlash))
```

Comment:
Potentially move into https://github.com/gatsbyjs/gatsby/blob/trailing-slash/packages/gatsby/src/commands/serve.ts#L297 and see if we can mimic express.static

Reply:
Have something in static/something.html and /something shouldn't be redirected.